doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java

```diff
@@ -320,9 +337,13 @@ public void renderDocument(
             String resource = doc.getAbsolutePath();
 
             Parser parser = doxia.getParser(docRenderingContext.getParserId());
-            // DOXIASITETOOLS-146 don't render comments from source markup
-            parser.setEmitComments(false);
-
+            ParserConfiguration parserConfiguration = docRenderingContext.getParserConfiguration();
+            if (parserConfiguration != null) {
+                parserConfiguration.accept(parser);
+            } else {
+                // DOXIASITETOOLS-146 don't render comments from source markup
+                parser.setEmitComments(false);
```

Comment:
TODO: once https://github.com/apache/maven-doxia/pull/180 is merged, the default for emitting anchors should be changed as well.